games-fps/yamagi-quake2/patches/yamagi_quake2-7.30.patchset

```diff
@@ -86,3 +86,45 @@ index a2e6f18..9397d97 100644
 -- 
 2.19.1
 
+From d0c494713cad012e25714e676bdb2eb69395a9e4 Mon Sep 17 00:00:00 2001
+From: Nikolay0054 <nikolay0054@gmail.com>
+Date: Fri, 2 Nov 2018 23:37:51 +0500
+Subject: [PATCH] fix find path
+
+---
+ src/backends/generic/misc.c | 13 +++++++++++++
+ 1 file changed, 13 insertions(+)
+
+diff --git a/src/backends/generic/misc.c b/src/backends/generic/misc.c
+index 973a2cba..73ba13f2 100644
+--- a/src/backends/generic/misc.c
++++ b/src/backends/generic/misc.c
+@@ -45,6 +45,10 @@
+ #include <mach-o/dyld.h> // _NSGetExecutablePath
+ #endif
+ 
++#ifdef __HAIKU__
++#include <FindDirectory.h> // find_path()
++#endif
++
+ #ifndef PATH_MAX
+ // this is mostly for windows. windows has a MAX_PATH = 260 #define, but allows
+ // longer paths anyway.. this might not be the maximum allowed length, but is
+@@ -119,6 +123,15 @@ static void SetExecutablePath(char* exePath)
+ 	// TODO: realpath() ?
+ 	// TODO: no idea what this is if the executable is in an app bundle
+ 
++#elif defined(__HAIKU__)
++
++	size_t bufSize = PATH_MAX;
++	
++	if(find_path(B_APP_IMAGE_SYMBOL, B_FIND_PATH_IMAGE_PATH, NULL, exePath, bufSize) != B_OK)
```

Comment:
`PATH_MAX` should be put directly in place of `bufSize`, since `find_path()` takes a value directly.